src/BuiltInTools/dotnet-watch/HotReload/RestartPrompt.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Tasks;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher
+{
+    internal sealed class RestartPrompt(IReporter reporter, ConsoleInputReader requester, bool? noPrompt)
+    {
+        private bool? _restartImmediatelySessionPreference = noPrompt;
+
+        public async ValueTask<bool> WaitForRestartConfirmationAsync(string question, CancellationToken cancellationToken)
+        {
+            if (_restartImmediatelySessionPreference.HasValue)
+            {
+                reporter.Output("Restarting");
+                return _restartImmediatelySessionPreference.Value;
+            }
+
+            var key = await requester.GetKeyAsync(
+                $"{question} Yes (y) / No (n) / Always (a) / Never (v)",
+                AcceptKey,
+                cancellationToken);
+
+            switch (key)
```

Comment:
Any concern about keyboard modifiers (ctrl/alt/shift) in this code?

Reply:
Have not changed the behavior here - but I guess it should check that modifiers are not active.